clang/lib/Sema/SemaExpr.cpp

```diff
@@ -17257,8 +17258,11 @@ void
 Sema::PushExpressionEvaluationContext(
     ExpressionEvaluationContext NewContext, ReuseLambdaContextDecl_t,
     ExpressionEvaluationContextRecord::ExpressionKind ExprContext) {
-  Decl *ClosureContextDecl = ExprEvalContexts.back().ManglingContextDecl;
-  PushExpressionEvaluationContext(NewContext, ClosureContextDecl, ExprContext);
+  const auto &PrevRec = ExprEvalContexts.back();
+  PushExpressionEvaluationContext(NewContext, PrevRec.ContextDecl,
+                                  PrevRec.ContextArgs, ExprContext);
+  ExprEvalContexts.back().HasReusedDeclContext = true;
+  ExprEvalContexts.back().LazyContextDeclPos = PrevRec.LazyContextDeclPos;
```

Comment:
FYI, `PrevRec` would be invalidated after `PushExpressionEvaluationContext()`, so you probably want to save the value before the call.

This caused `clang/test/SemaTemplate/stack-exhaustion.cpp` to fail on my machine.

Reply:
Yeah, I saw this failure on CI, but not on my machine (MacOS).

Thanks for taking a look.